plutus-core/src/Language/PlutusCore/Lexer.x

```diff
@@ -47,31 +48,71 @@ import Text.Read (readMaybe)
    where we can use keywords as variable names. A similar strategy is
    used for built in type names. -}
 
+{- Note [Literal Constants]
+   For literal constants, we accept certain types of character sequences that are
+   then passed to user-defined parsers which convert them to built-in constants.
+   Literal constants have to specify the type of the constant, so we have (con
+   integer 9), (con string "Hello"), and so on.  This allows us to use the same
+   literal syntax for different types (eg, integer, short, etc) and shift most
+   of the responsibility for parsing constants out of the lexer and into the
+   parser (and eventually out of the parser to parsers supplied by the types
+   themselves).
+
+   We allow:
+    * ()
+    * Single-quoted printable characters
+    * Double-quoted sequences of printable characters
+    * Unquoted sequences of printable characters not including '(' and ')'
```

Comment:
Could we just say we allow literally anything (printable) up to a closing `)`? I don't think we need to allow `()` for unit, having to say `(con unit unitval)` would frankly be better than overloading the parens.

Reply:
No.  There are cases when you have a `)` that's not closing anything.  What if we have a literal string containing ')', for example?

Reply:
>  I don't think we need to allow `()` for unit, having to say `(con unit unitval)` would frankly be better than overloading the parens.

That would involve altering the `PrettyConst` instance for `()` (not a very onerous task, but it would make it do something different from what Haskell does).  If we do that then I propose we change the instance for `Bool` to print `true` and `false` while we're at it.